core/storage/classroom/gae_models.py

```diff
@@ -0,0 +1,177 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing classroom data."""
+
+from __future__ import annotations
+
+from core import utils
+from core.platform import models
+import core.storage.base_model.gae_models as base_models
+
+from typing import Dict, List, Optional
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class ClassroomModel(base_models.BaseModel):
+    """Model to store the classroom data.
+
+    The id of instances of this class is in the form of random hash of
+    12 chars.
+    """
+
+    # The name of the classroom.
+    name = datastore_services.StringProperty(required=True, indexed=True)
+    # The url fragment of the classroom.
+    url_fragment = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # A text to provide course details present in the classroom.
+    course_details = datastore_services.StringProperty(
+        default='', indexed=True, required=False)
```

Comment:
Could you drop the default for this and have required=True? Ditto for the next one.

Reply:
Updated